### ISSUE TITLE
improve modal screen (sheet UI)

### ISSUE THREAD

**Issue by flexsurfer** [status-im/status-react#15546]
we have an option to open modal screen with sheet UI

```clojure
{:name      :community-requests-to-join
 :options   {:sheet? true
             :insets {:bottom? true}}
 :component join-menu/request-to-join}

rf/dispatch [:open-modal :community-requests-to-join]
```

implementation is here: `status-im2.common.bottom-sheet-screen.view` https://github.com/status-im/status-mobile/blob/6417d90e1c15e9d51c0cd851d8a9a53d189588a2/src/status_im2/common/bottom_sheet_screen/view.cljs#L49

it needs some improvements
- [x] 1) currently modal screen has its own transition animation, so we have two animations, we need to disable navigation animation
- [x] 2) it uses gesture library but some screens use react native section list which is not supported buy gesture library, so this should fixed somehow, currently gesture was moved to handle view only
- [x] 3) status bar should be behind the backdrop, currently its light
- [ ] 4) there are some screen with scrolling states
 
<img width="625" alt="image" src="https://user-images.githubusercontent.com/11790366/229072782-cbc3da9f-465e-4872-afca-159fd82d9247.png">

5) would be great to have a video example from the design team on how the transition should work cc @pedro-et @xAlisher 

- [ ] 6) this state looks wrong, when we pull the screen down there is no backdrop

<img width="337" alt="image" src="https://user-images.githubusercontent.com/11790366/229073689-a3a3a96e-61af-484f-bdc5-842a264a7526.png">

cc @OmarBasem

**Comment by OmarBasem** [comment]
Thanks for opening this issue @flexsurfer

1. Can you elaborate what do you mean by this point
2. We need to use gesture/flat-list. SectionList actually uses FlatList under the hood to create sections. We need to modify the data and render method to create sections ourselves. Or better, we can make a reusable gesture section list component that uses the same API as SectionList.
3. Hmm..
4. Can you elaborate?
5. That would be great
6. Not sure what is this referring to

**Comment by flexsurfer** [comment]
1- screen container is moving, so when modal is opened or closed you can see how backgdrop is moving, it shouldn't move, it should have only opacity
4- it's something complex, so the sheet should have states, first it opened not in full screen, and then if you scroll it it moves to the fill screen
6- typo , there is no backdrop, and you can see it looks like sheet the part of the screen, there is no backdrop

**Comment by OmarBasem** [comment]
@flexsurfer to fix the issues 1, 3 and 4, the backdrop component would have to be separated from the `bottom-sheet-screen` component. It would have to be displayed as an overlay on top of the current content, then the modal screen opens. I tried using `show-overlay` from RNN, but it seems it will render the backdrop on top of everything. I tried to play with the z-index but it did not work. So probably the backdrop would have to be displayed as an overlay in another way.
